composeApp/src/main/kotlin/io/github/couchtracker/utils/DeferredConstant.kt

```diff
@@ -0,0 +1,25 @@
+package io.github.couchtracker.utils
+
+import kotlinx.coroutines.CompletableDeferred
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+private class DeferredConstant<T>(val initialize: suspend () -> T) {
```

Comment:
Have you considered using Koin lazy modules with background loading instead?

Docs: https://insert-koin.io/docs/reference/koin-core/lazy-modules/

You'll need to add this dependency:
```
koin-coroutines = { group = "io.insert-koin", name = "koin-core-coroutines" }
```